docs/source/reference/api/gen/oauthenticator.auth0.rst

```diff
@@ -0,0 +1,9 @@
+oauthenticator.auth0
+====================
+
+.. automodule:: oauthenticator.auth0
+
+
+.. autoconfigurable:: Auth0OAuthenticator
+
+.. autoconfigurable:: LocalAuth0OAuthenticator
```

Comment:
The `authenticator.rst.tpl` is used to generate these files, which are files that render nicely with the `autodoc_traits` extension.

Since we generate them, we don't need them to be comitted. So let's remove them and update the .gitignore file to reflect this folder was relocated and no longer gitignored.

Reply:
Also, update the `make devenv` entry in the Makefile / make.bat files as they use an `--ignore` flag to not rebuild because the generated files are updated by the build itself.

Reply:
Good idea to delete them, thank you! I've just added a commit that does that and also added a `clean` target, so that they get cleanup up on `make clean` too  https://github.com/jupyterhub/oauthenticator/pull/561/commits/60d7c07b00accb5a5e750dcf7b2e12d95eb2e28c

Reply:
Good idea to let `clean` support cleaning up those files.

The commit removed `--ignore` for sphinx-autobuild rather than updating it to reflect the location of files to not watch for changes, so when trying `make devenv` i got a flurry of rebuilds that never stopped.

Reply:
Ah, I understand! Fixing it now.

Reply:
Done @consideRatio! Had to include the `gen` directory too in the `ignore` glob https://github.com/jupyterhub/oauthenticator/pull/561/commits/6f09254fc72eeb12e3c2d0b361aff7b166d28d43